### DISCUSSION TITLE
create_index（）

### DISCUSSION THREAD

**Discussion by LMERose** [milvus-io/milvus#35249] [Q&A and General discussion]
hello, if I want to use this :create_collection --> insert（） --> create_index（） --> load（） --> search（）,what should I do?Which part of the program should I modify? I'm using the langchain interface

**Comment by xiaofan-luan** [comment]
Milvus support the process.
But unfortunately langchain might don't have that finegrained control

**Reply by LMERose** [reply]
tanks,I just give up use langchain to support the process.I choose write it by myself.

**Comment by xiaofan-luan** [comment]
@jaelgu 
could you help me on checking it

**Comment by jaelgu** [comment]
@LMERose In langchain_milvus, it will automatically create index for dense vector field and load collection when initiating the Milvus module. By default, it will use HNSW or AUTOINDEX for Zilliz Cloud service. You are able to use the parameter `index_params` to specify your own index. For example:
```python
from langchain_milvus.vectorstores import Milvus

# Create collection only without inserting data
# This will create index and load collection in advance, which will automatically index and load any data inserted in the future.
vector_db = Milvus(
    embeddings,
    connection_args={"uri": URI},
    collection_name="collection_1",
    # index_params={YOUR_INDEX_PARAMS}
)

# Insert data into collection
vector_db.add_documents(langchain_documents)

# Search
vector_db.similarity_search(query_text)
```
There are other options to insert and retrieve data via LangChain. See more code examples at https://python.langchain.com/v0.2/docs/integrations/vectorstores/milvus/
